localstorage/main.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * This is an example using LocalStorage in your main page to to keep track of
+ * whether its underlying web worker can force using persistence without
+ * breaking other tabs/web workers. Since only one tab can use persistence at a
+ * time, we're using LocalStorage to restrict persistence.
+ * 
+ * You can invoke Firestore functionality from your main page by passing messages
+ * to the web worker running Firebase, as shown in the enableNetwork/
+ * disableNetwork example.
+ */
+
+const dedicatedWorker = new Worker("firebase-worker.js");
+
+function main() {
+  const persistenceAvailable = isPersistenceAvailable();
+  
+  // Whether the current tab is using persistence.
+  let isUsingPersistence = false;
+
+  // Tell the Firestore worker if it can enable persistence.
+  if (persistenceAvailable) {
+    dedicatedWorker.postMessage('persistenceAvailable');
+  } else {
+    dedicatedWorker.postMessage('persistenceUnavailable');
+  }
+
+  dedicatedWorker.onmessage = function(event) {
+    // Mark persistence as unavailable to other tabs, if this tab has 
+    // successfully enabled persistence.
+    if (event.data === 'takePersistence') {
+      localStorage.setItem('persistenceAvailable', 'false');
+      isUsingPersistence = true;
+    }
+  }
+
+  // If the worker in this tab is using persistence, release persistence
+  // before the page terminates. This allows the next worker that loads
+  // to use persistence. 
+  window.addEventListener("beforeunload", function() {
+    console.log('Terminating. Releasing persistence.');
+    if (isUsingPersistence) {
+      localStorage.setItem('persistenceAvailable', 'true');
+    }
+  })
+}
+
+/**
+ * Checks if persistence is available in LocalStorage, or initializes the
+ * field if it has not yet been set.
```

Comment:
You might want to add some sort of NOTE or TODO comment somewhere that mentions that if the page crashes, persistence could be permanently marked unavailable in LocalStorage and so it might be a good idea to implement some sort of time-based expiration as a fallback.

(i.e. hint at the kind of logic we do in our lease tracking but without going to the trouble of implementing it)